arelle/ModelValue.py

```diff
@@ -289,79 +343,100 @@ def __new__(cls, y, m, d, hr=0, min=0, sec=0, microsec=0, tzinfo=None, dateOnly=
         dateTime.dateOnly = dateOnly
         return dateTime
-    def __copy__(self):
+    def __copy__(self) -> DateTime:
         return DateTime(self.year, self.month, self.day, self.hour, self.minute, self.second, self.microsecond, self.tzinfo, self.dateOnly)
-    def __str__(self):
+    def __str__(self) -> str:
         # note does not print negative dates but xml does allow negative date
         tz = tzinfoStr(self)
         if self.dateOnly:
             return "{0.year:04}-{0.month:02}-{0.day:02}{1}".format(self, tz)
         else:
             return "{0.year:04}-{0.month:02}-{0.day:02}T{0.hour:02}:{0.minute:02}:{0.second:02}{1}".format(self, tz)
-    def addYearMonthDuration(self, other, sign):
+    def addYearMonthDuration(self, other: YearMonthDuration, sign: int) -> DateTime:
         m = self.month + sign * other.months - 1 # m is zero based now (0 - Jan, 11 - Dec)
         y = self.year + sign * other.years + m // 12
         m = (m % 12) + 1 # m back to 1 based (1 = Jan)
         d = self.day
         lastDay = lastDayOfMonth(y, m)
         if d > lastDay: d = lastDay
         return DateTime(y, m, d, self.hour, self.minute, self.second, self.microsecond, self.tzinfo, self.dateOnly)
-    def __add__(self, other):
+    def __add__(self, other: YearMonthDuration | Time) -> DateTime: # type: ignore[override]
         if isinstance(other, YearMonthDuration):
             return self.addYearMonthDuration(other, 1)
         else:
-            if isinstance(other, Time): other = dayTimeDuration(other)
-            dt = super(DateTime, self).__add__(other)
+            if isinstance(other, Time):
+                _other: DayTimeDuration = dayTimeDuration(other)
+            dt = super(DateTime, self).__add__(_other)
             return DateTime(dt.year, dt.month, dt.day, dt.hour, dt.minute, dt.second, dt.microsecond, dt.tzinfo, self.dateOnly)
 
-    def __sub__(self, other):
+    @overload
+    def __sub__(self, other: YearMonthDuration) -> DateTime: ...
+
+    @overload
+    def __sub__(self, other: datetime.timedelta) -> DayTimeDuration: ...
+
+    @overload
+    def __sub__(self, other: datetime.datetime) -> DateTime: ...
+
+    def __sub__(self, other: YearMonthDuration | datetime.timedelta | datetime.datetime) -> DateTime | DayTimeDuration:
         if isinstance(other, YearMonthDuration):
             return self.addYearMonthDuration(other, -1)
         else:
-            dt = super(DateTime, self).__sub__(other)
-            if isinstance(dt,datetime.timedelta):
+            dt = super(DateTime, self).__sub__(other) # type: ignore[operator]
+            if isinstance(dt, datetime.timedelta):
                 return DayTimeDuration(dt.days, 0, 0, dt.seconds)
-            else:
+
+            if isinstance(dt, datetime.datetime):
+                # Note 2022-09-10:
+                # Is this a bug? Why are we assigning to other but never use it?
```

Comment:
Is this a bug?

Reply:
I think so. It appears that if `other` is `Time` it will throw a not implmeneted error

Reply:
I think so. This cold is very old, but appears to have been a copy paste from `__add__` above

Reply:
May I remove it in this PR?

Reply:
Yes, please.